contracts/gateway/OSTComposer.sol

```diff
@@ -0,0 +1,436 @@
+            "Stake amount must not be zero."
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be zero."
+        );
+
+        require(
+            stakeRequestHashes[msg.sender][address(_gateway)] == bytes32(0),
+            "Request for this staker at this gateway is already in process."
+        );
+
+        stakeRequestHash_ = hashStakeRequest(
+            msg.sender,
+            address(_gateway),
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            _nonce
+        );
+
+        stakeRequestHashes[msg.sender][address(_gateway)] = stakeRequestHash_;
+
+        StakerProxy stakerProxy = stakerProxies[msg.sender];
+
+        if(address(stakerProxy) == address(0)) {
+            stakerProxy = new StakerProxy(msg.sender);
+            stakerProxies[msg.sender] = stakerProxy;
+        }
+
+        require(
+            _gateway.getNonce(address(stakerProxy)) == _nonce,
+            "Incorrect staker nonce."
+        );
+
+        stakeRequests[stakeRequestHash_] = StakeRequest({
+            amount: _amount,
+            beneficiary: _beneficiary,
+            gasPrice: _gasPrice,
+            gasLimit: _gasLimit,
+            nonce: _nonce,
+            staker: msg.sender,
+            gateway: _gateway
+        });
+        activeStakeRequestCount[msg.sender] = activeStakeRequestCount[msg.sender].add(1);
+
+        EIP20Interface valueToken = _gateway.valueToken();
+
+        require(
+            valueToken.transferFrom(msg.sender, address(this), _amount),
+            "Value token transfer returned false."
+        );
+
+        emit StakeRequested(
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            _nonce,
+            msg.sender,
+            address(_gateway),
+            stakeRequestHash_
+        );
+    }
+
+    /**
+     * @notice Facilitator calls the method to initiate the stake process.
+     *         Staked amount from composer and bounty amount from facilitator
+     *         is then transferred to the StakerProxy contract of the staker.
+     *
+     * @param _stakeRequestHash Unique hash for the stake request which is to
+     *                          be Processed.
+     * @param _hashLock Hashlock provided by the facilitator.
+     *
+     * @return messageHash_ Hash unique for each request.
+     */
+    function acceptStakeRequest(
+        bytes32 _stakeRequestHash,
+        bytes32 _hashLock
+    )
+        external
+        mutex
+        onlyWorker
+        returns(bytes32 messageHash_)
+    {
+        StakeRequest storage stakeRequest = stakeRequests[_stakeRequestHash];
+        require(
+            stakeRequest.staker != address(0),
+            "Stake request must exists."
+        );
+
+        EIP20GatewayInterface gateway = stakeRequest.gateway;
+
+        StakerProxy stakerProxy = stakerProxies[address(stakeRequest.staker)];
```

Comment:
Doesn't this work? `stakeRequest.staker` is stored as `address` in the Struct.

```suggestion
        StakerProxy stakerProxy = stakerProxies[stakeRequest.staker];
```

Reply:
Thanks. It works.
Changed it.